03_Modules/Locations/src/module/admin.ts

```diff
@@ -0,0 +1,172 @@
+import { Service } from 'typedi';
+import { type ILogObj, Logger } from 'tslog';
+import {
+  ChargingStation,
+  Location,
+  SequelizeLocationRepository,
+} from '@citrineos/data';
+import {
+  AdminConnectorDTO,
+  AdminEVSEDTO,
+  AdminLocationDTO,
+  CitrineOcpiLocationMapper,
+  LocationDTO,
+  LocationsBroadcaster,
+  OcpiLocation,
+  OcpiLocationProps,
+  OcpiLocationRepository,
+  OcpiEvse,
+  OcpiEvseRepository,
+  OcpiConnector,
+  OcpiConnectorRepository,
+  VariableAttributesUtil,
+  OcpiLocationsUtil,
+  InvalidParamException,
+} from '@citrineos/ocpi-base';
+
+@Service()
+export class AdminLocationsService {
+  constructor(
+    private logger: Logger<ILogObj>,
+    private locationMapper: CitrineOcpiLocationMapper,
+    private locationRepository: SequelizeLocationRepository,
+    private ocpiLocationRepository: OcpiLocationRepository,
+    private ocpiEvseRepository: OcpiEvseRepository,
+    private ocpiConnectorRepository: OcpiConnectorRepository,
+    private locationsBroadcaster: LocationsBroadcaster,
+    private variableAttributesUtil: VariableAttributesUtil,
+    private ocpiLocationsUtil: OcpiLocationsUtil,
+  ) {}
+
+  public async createOrUpdateLocation(
+    adminLocationDto: AdminLocationDTO,
+    broadcast: boolean,
+  ): Promise<LocationDTO> {
+    this.logger.debug(
+      adminLocationDto.id
+        ? `Updating Location ${adminLocationDto.id}`
+        : `Creating Location ${adminLocationDto.name}`,
+    );
+
+    const [invalidLocation, message] = AdminLocationDTO.IS_LOCATION_INVALID(adminLocationDto);
+    if (invalidLocation) {
+      throw new InvalidParamException(message ?? 'Invalid location request.');
+    }
+
+    const [ocpiLocation, citrineLocation] =
+      this.mapAdminLocationDtoToEntities(adminLocationDto);
+    const savedCitrineLocation =
+      await this.locationRepository.createOrUpdateLocationWithChargingStations(
+        citrineLocation,
+      );
+    ocpiLocation[OcpiLocationProps.citrineLocationId] = savedCitrineLocation.id;
```

Comment:
Starting to question if this "Props" convention is a good move, as it seems to make readability a bit more difficult imo.

Reply:
i think they're good for some cases (like if you're constructing a query and you want to ensure the query has the right property), but maybe not so good for cases where you're just trying to grab that property from the object, since the IDE should be able to tell you if the property is real or not. 

i've updated general usages in this PR to follow those rules (only include for queries, don't include when doing object assignments/grabbing a property since we already know those properties are legit)